crates/game/pong/src/game.rs

```diff
@@ -88,6 +100,36 @@ impl Game for PongGame {
             self.reset_ball()
         }
     }
+
+    fn tick_model(&mut self) {
+        if let PongPlayerInput::Model(model) = &mut self.player.0.input {
+            let input = vec![
+                self.player.0.pos as f64,
+                self.state.ball_pos.x as f64,
+                self.state.ball_pos.y as f64,
+                self.state.ball_dir.x as f64,
+                self.state.ball_dir.y as f64,
+            ];
+
+            model.tick(Some(input));
+        }
+
+        if let PongPlayerInput::Model(model) = &mut self.player.1.input {
+            let input = vec![
+                self.player.1.pos as f64,
+                self.state.ball_pos.x as f64,
+                self.state.ball_pos.y as f64,
+                self.state.ball_dir.x as f64,
+                self.state.ball_dir.y as f64,
+            ];
+
+            model.tick(Some(input));
+        }
+    }
+
+    fn score(&self) -> f32 {
+        self.state.score.0 as f32 - self.state.score.1 as f32
+    }
```

Comment:
An feadback function should also be called after each game tick or something line that. Should be use to give penalties or rewards for certain actions. Not needed for now in pong.

Also here you could weight points of enemies more to give bigger penalties

Reply:
> Also here you could weight points of enemies more to give bigger penalties

Definitely something we can try

> An feadback function should also be called after each game tick or something line that. Should be use to give penalties or rewards for certain actions. Not needed for now in pong.

in which way is feedback during a simulation need?
could this simply be done with a more complex score function which e.g. also uses the number of successful defenses?

Reply:
Eg an roboter learns to walk. Per second it dhould get an reward depending on how up straight it is currently walking